api.go

```diff
@@ -737,10 +737,8 @@ func (r *Raft) Apply(cmd []byte, timeout time.Duration) ApplyFuture {
 func (r *Raft) ApplyLog(log Log, timeout time.Duration) ApplyFuture {
 	metrics.IncrCounter([]string{"raft", "apply"}, 1)
 
-	var timer <-chan time.Time
-	if timeout > 0 {
-		timer = time.After(timeout)
-	}
+	timer := newTimer(timeout)
+	defer timer.Stop()
```

Comment:
I think we can accomplish the same thing without wrapping `time.Timer`

```suggestion
	var timer <-chan time.Time
	if timeout > 0 {
		t := time.Newtimer(timeout)
		defer t.Stop()
		timer = t.C
	}
```

This is a few extra lines, but may actually result in a smaller diff. I think in the long term not having a wrapper around `time.Timer` will be easier to maintain.

Reply:
When reviewing this PR, I had to pause and locally test the behaviour of `timerWrapper`  to gain confidence that it would behave how I expect (and found a potential bug with `(timerWrapper).Reset`).

From a maintainer's perspective, I think code duplication is an acceptable cost here. The wrapper certainly saves many lines of code but may increase developer time spent reading and understanding the wrapper type.

I'd like to request that we go with @dnephin's suggestion for using the std lib directly